Implementacion group choice nuevo modelo

LuisParedes1 opened the pull request.

gilopezgiles committed:
posible solucion para preguntas especiales

LuisParedes1 committed:
Implementacion de GroupChoice

Se implemento group choice con el nuevo modelo planteado (usando T generic)

LuisParedes1 committed:
Fix: Arregle problema del orden de las preguntas

Cuando las preguntas esten desordenadas pero en el grupo correcto lo tomara como correcta

LuisParedes1 committed:
Merge branch 'develop' into implementacion-GroupChoice-Nuevo-Modelo

LuisParedes1 committed:
Merging Fixes

Se arreglaron conflictos entre el modelo viejo y el nuevo

LuisParedes1 committed:
Se agregaron pruebas para multiplicador

gylopezgiles commented on src/main/java/edu/fiuba/algo3/modelo/pregunta/Preguntable.java:
```diff
@@ -1,17 +1,16 @@
+    int establecerPuntuacion(T seleccion) throws MultiplicadorExcepcion;
 
-    int establecerPuntuacion(List<Opcion> opciones, MultiplicableStrategy multiplicador) throws MultiplicadorExcepcion;
+    int establecerPuntuacion(T opciones, MultiplicableStrategy multiplicador) throws MultiplicadorExcepcion;
```
habria que ponerle tmb seleccion como en el de arriba :)

gylopezgiles commented on src/main/java/edu/fiuba/algo3/modelo/pregunta/multiplechoice/GroupChoice.java:
```diff
@@ -0,0 +1,95 @@
+    private final String pregunta;
+    private final List<Opcion> opciones;
+    private List<Opcion> opcionesVerdaderas;
+    private List<Opcion> opcionesFalsas;
```
cambiar el nombre a algo como opcionesPrimerGrupo, opcionesSegundoGrupo, ya que por mas que usemos el boolean para separar en 2 grupos no son opciones verdaderas o falsas

gylopezgiles commented on src/main/java/edu/fiuba/algo3/modelo/pregunta/multiplechoice/GroupChoice.java:
```diff
@@ -0,0 +1,95 @@
+        }
+
+        List<Opcion> respuestasGrupoVerdadero = grupos.get(0);
+        List<Opcion> respuestasGrupoFalso = grupos.get(1);
```
idem a opciones, cambiaria el nombre a respuestasPrimerGrupo, respuestasSegundoGrupo o algo asi :)

gylopezgiles commented on src/main/test/edu/fiuba/algo3/modelo/pregunta/multiplechoice/GroupChoiceTest.java:
```diff
@@ -0,0 +1,279 @@
+
+        String preguntaTexto = "¿Cuales ciudades pertenecen a Nueva Zelanda y cuales no?";
+
+        Boolean pertenece = Boolean.TRUE;
```
capaz quedaria mas claro un Boolean pertenecePrimerGrupo y despues pones !pertenecePrimerGrupo en las que van al otro grupo

gylopezgiles commented on src/main/java/edu/fiuba/algo3/modelo/pregunta/CreadorPregunta.java:
```diff
@@ -8,11 +8,9 @@
+    public static <T> Preguntable crearPregunta(TipoPregunta tipoPregunta, String pregunta, T opciones) throws ParametrosInvalidosExcepcion, TipoPreguntaNoImplementadaException {
         switch (tipoPregunta) {
             case VerdaderoFalsoClasico:
                 return new VerdaderoFalsoClasico(pregunta, opciones);
```
falta agregar al switch el groupchoice y al tipoPregunta y su test

LuisParedes1 committed:
Fix: Se arreglo el creador de Pregunta

Ahora incluye en el enum el GroupChoice y se agrego la prueba que verifica que este bien

LuisParedes1 committed:
Arreglo minimo

LuisParedes1 committed:
Merge branch 'develop' into implementacion-GroupChoice-Nuevo-Modelo
